src/neo4j/_async/work/result.py

```diff
@@ -242,17 +280,81 @@ def on_success(summary_metadata):
             n=-1,
             qid=self._qid,
             on_success=on_success,
             on_failure=on_failure,
-            on_summary=on_summary,
         )
         self._streaming = True
 
+    def _handle_warnings(self) -> None:
+        sev_filter: t.Tuple[NotificationMinimumSeverity, ...] = ()
+        if (
+            self._warn_notification_severity
+            == NotificationMinimumSeverity.WARNING
+        ):
+            sev_filter = (NotificationMinimumSeverity.WARNING,)
+        elif (
+            self._warn_notification_severity
+            == NotificationMinimumSeverity.INFORMATION
+        ):
+            sev_filter = (NotificationMinimumSeverity.INFORMATION,
+                          NotificationMinimumSeverity.WARNING)
+
+        summary = self._obtain_summary()
+        query = self._metadata.get("query")
+        for notification in summary.summary_notifications:
+            log_call = notification_log.debug
+            if notification.severity_level == NotificationSeverity.INFORMATION:
+                log_call = notification_log.info
+            elif notification.severity_level == NotificationSeverity.WARNING:
+                log_call = notification_log.warning
+            log_call(
+                "Received notification from DBMS server: %s",
+                NotificationPrinter(notification, query, one_line=True)
+            )
+
+            if notification.severity_level not in sev_filter:
+                continue
+            warning_cls: te.Type[Warning] = Neo4jWarning
+            if notification.category == NotificationCategory.DEPRECATION:
+                warning_cls = Neo4jDeprecationWarning
+            creation_frame = self._creation_frame
+            if creation_frame is False:
+                warn(warning_cls(notification), stacklevel=1)
+            else:
+                globals_ = creation_frame.frame.f_globals
+                warning_registry = globals_.get("__warningregistry__", {})
+                warn_explicit(
+                    warning_cls(notification, query),
+                    None,
+                    creation_frame.filename,
+                    creation_frame.lineno,
+                    module=globals_["__name__"],
+                    registry=warning_registry,
+                    module_globals=globals_
+                )
+
+    @property
+    def _creation_frame(self) -> t.Union[t.Literal[False], inspect.FrameInfo]:
+        if self._creation_frame_cache is not None:
+            return self._creation_frame_cache
+        if self._creation_stack is None:
+            self._creation_frame_cache = False
+        else:
+            self._creation_frame_cache = next(
+                (frame for frame in self._creation_stack
+                 if not frame.filename.startswith(str(_driver_dir))),
+                False
+            )
+        assert self._creation_frame_cache is not None  # help mypy a little
+        return self._creation_frame_cache
```

Comment:
This will throw if self._creation_stack is None.

Reply:
I don't quit understand how. If `self._creation_stack` is `None`, then we end up in this branch 

https://github.com/neo4j/neo4j-python-driver/blob/2be572fe7a915255a7b046b31720b7d34b7b9e74/src/neo4j/_async/work/result.py#L347-L348

So `self._creation_frame_cache is not None` evaluates to `False` and we also return `False`.

Reply:
Sorry, my mind tricked me.